src/Form/AdminSettingsForm.php

```diff
@@ -346,30 +345,29 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
   private function setCredentialValues(Config $settings, array $values) {
     $settings->set('credential.account_name', $values['account_name']);
     $settings->set('credential.customer_site', $values['customer_site']);
-    $settings->set('credential.api_url', $values['api_url']);
-    $settings->set('credential.access_key', $values['access_key']);
-    if (!empty($values['secret_key'])) {
-      $settings->set('credential.secret_key', $values['secret_key']);
+    $settings->set('credential.assets_url', 'https://' . $this->removeProtocol($values['assets_url']));
+
+    $settings->clear('credential.api_url');
+    if (!empty($values['api_url'])) {
+      $settings->set('credential.api_url', 'https://' . $this->removeProtocol($values['api_url']));
     }
-    $settings->set('credential.js_path', $values['js_path']);
-    $settings->set('credential.assets_url', $values['assets_url']);
     $settings->clear('credential.oauth_url');
     if (!empty($values['oauth_url'])) {
-      $settings->set('credential.oauth_url', $values['oauth_url']);
+      $settings->set('credential.oauth_url', 'https://' . $this->removeProtocol($values['oauth_url']));
     }
   }
 
   /**
-   * Remove the protocal from an URL string.
+   * Remove the protocol and trailing slashes from an URL string.
    *
    * @param string $url
    *   URL.
    *
    * @return string
-   *   Same URL as input except with the 'http://' and 'https://' trimmed.
+   *   Same URL as input except with the 'http://' and 'https://' and trailing '/' trimmed.
    */
-  private function removeProtocal($url) {
-    return preg_replace('~^https?://~', '', $url);
+  private function removeProtocol($url) {
+    return rtrim(preg_replace('~^https?://~', '', $url), '/');
```

Comment:
Are you sure this is the best approach? I seem to remember that drupal or symfony has helper functions to aid you with such transformations? what if folks for some weird reason make a typo and make it htts:// ? :)

For example, the link module does it like this:
```
// We drop any scheme of the url.
$scheme = parse_url($url);
$search = '/' . preg_quote($scheme['scheme'] . '://', '/') . '/';
$replace = '';
$display_url = preg_replace($search, $replace, $url, 1);
```